$NetBSD: patch-ac,v 1.2 2013/09/13 09:11:13 jperkin Exp $

Avoid "error: call of overloaded 'pow(int, unsigned int)' is ambiguous"

--- plugins/sine.cpp.orig	2007-11-06 10:13:34.000000000 +0000
+++ plugins/sine.cpp
@@ -62,12 +62,21 @@ initialise_sine_table() {
 	g_pfSineTable[lIndex] = LADSPA_Data(sin(dShift * lIndex));
   }
   if (g_fPhaseStepBase == 0) {
-    g_fPhaseStepBase = (LADSPA_Data)pow(2, sizeof(unsigned long) * 8);
+    g_fPhaseStepBase = (LADSPA_Data)pow((float)2, (float)sizeof(unsigned long) * 8);
   }
 }
 
 /*****************************************************************************/
 
+LADSPA_Handle instantiateSineOscillator(const LADSPA_Descriptor *, unsigned long SampleRate);
+void connectPortToSineOscillator(LADSPA_Handle Instance, unsigned long Port, LADSPA_Data * DataLocation);
+void activateSineOscillator(void * pvHandle);
+void runSineOscillator_FreqAudio_AmpAudio(LADSPA_Handle Instance, unsigned long SampleCount);
+void runSineOscillator_FreqAudio_AmpCtrl(LADSPA_Handle Instance, unsigned long SampleCount);
+void runSineOscillator_FreqCtrl_AmpAudio(LADSPA_Handle Instance, unsigned long SampleCount);
+void runSineOscillator_FreqCtrl_AmpCtrl(LADSPA_Handle Instance, unsigned long SampleCount);
+void cleanupSineOscillator(void *pvHandle);
+
 class SineOscillator {
 private:
 
